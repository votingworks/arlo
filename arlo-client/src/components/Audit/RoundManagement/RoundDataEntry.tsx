import React from 'react'
import H2Title from '../../Atoms/H2Title'

const RoundDataEntry = () => {
  return <H2Title>Round 1 Data Entry</H2Title>
}

export default RoundDataEntry
