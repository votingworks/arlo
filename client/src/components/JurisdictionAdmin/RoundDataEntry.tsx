import React from 'react'
import { useParams } from 'react-router-dom'
import { H5, H3 } from '@blueprintjs/core'
import { Field, Formik, FormikProps } from 'formik'
import styled from 'styled-components'
import useContestsJurisdictionAdmin from './useContestsJurisdictionAdmin'
import useResults, { IResultValues } from './useResults'
import FormButton from '../Atoms/Form/FormButton'
import { IRound } from '../AuditAdmin/useRoundsAuditAdmin'
import { testNumber } from '../utilities'
import FormField from '../Atoms/Form/FormField'

const Contest = styled.div`
  padding: 10px 0;
`

const BottomButton = styled(FormButton)`
  margin: 30px 0;
`

const BlockLabel = styled.label`
  display: block;
  margin: 10px 0;
`

interface IProps {
  round: IRound
}

interface IValues {
  results: IResultValues
}

const RoundDataEntry: React.FC<IProps> = ({ round }) => {
  const { electionId, jurisdictionId } = useParams<{
    electionId: string
    jurisdictionId: string
  }>()
  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
  const [results, updateResults] = useResults(
    electionId,
    jurisdictionId,
    round.id
  )

  if (!results || !contestsQuery.isSuccess) return null
  const contests = contestsQuery.data

  const alreadySubmittedResults = Object.values(results).some(a =>
    Object.values(a).some(b => b)
  )

  const submit = async (values: IValues) => {
    updateResults(values.results)
  }

  return (
    <Formik initialValues={{ results }} enableReinitialize onSubmit={submit}>
      {({ handleSubmit, isSubmitting }: FormikProps<IValues>) => (
        <form>
          <H3>Enter Tallies</H3>
          <p>
            When you have examined all the ballots assigned to you, enter the
            number of votes recorded for each candidate/choice from the audited
            ballots.
          </p>
          {contests.map(contest => (
            <Contest key={contest.id}>
              <H5>{contest.name}</H5>
              {contest.choices.map(choice => (
                <BlockLabel
                  key={choice.id}
                  htmlFor={`results[${contest.id}][${choice.id}]`}
                >
                  Votes for {choice.name}:
                  <Field
                    id={`results[${contest.id}][${choice.id}]`}
                    name={`results[${contest.id}][${choice.id}]`}
                    disabled={alreadySubmittedResults}
                    validate={testNumber()}
                    component={FormField}
                  />
                </BlockLabel>
              ))}
            </Contest>
          ))}
          <BottomButton
            type="submit"
            intent="primary"
            loading={isSubmitting}
            disabled={alreadySubmittedResults}
            onClick={handleSubmit}
          >
            {alreadySubmittedResults ? `Tallies Submitted` : `Submit Tallies`}
          </BottomButton>
        </form>
      )}
    </Formik>
  )
}

export default RoundDataEntry
