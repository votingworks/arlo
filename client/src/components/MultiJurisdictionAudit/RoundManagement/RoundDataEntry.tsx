import React from 'react'
import { useParams } from 'react-router-dom'
import { H5 } from '@blueprintjs/core'
import { Field, Formik, Form, FormikProps } from 'formik'
import styled from 'styled-components'
import useContestsJurisdictionAdmin from './useContestsJurisdictionAdmin'
import { IRound } from '../useRoundsJurisdictionAdmin'
import Card from '../../Atoms/SpacedCard'
import FormField from '../../Atoms/Form/FormField'
import FormButton from '../../Atoms/Form/FormButton'
import { testNumber } from '../../utilities'
import useResults, { IResultValues } from './useResults'

const BottomButton = styled(FormButton)`
  margin: 30px 0;
`

const BlockLabel = styled.label`
  display: block;
  margin: 20px 0;
`

interface IProps {
  round: IRound
}

interface IValues {
  results: IResultValues
}

const RoundDataEntry = ({ round }: IProps) => {
  const { electionId, jurisdictionId } = useParams<{
    electionId: string
    jurisdictionId: string
  }>()
  const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
  const [results, updateResults] = useResults(
    electionId,
    jurisdictionId,
    round.id
  )

  if (!results || !contests) return null

  const alreadySubmittedResults = Object.values(results).some(a =>
    Object.values(a).some(b => b)
  )

  const submit = async (values: IValues) => {
    updateResults(values.results)
  }

  return (
    <Formik initialValues={{ results }} enableReinitialize onSubmit={submit}>
      {({ handleSubmit }: FormikProps<IValues>) => (
        <Form>
          <p>
            When you have examined all the ballots assigned to you, enter the
            number of votes recorded for each candidate/choice from the audited
            ballots.
          </p>
          {contests.map(contest => (
            <Card key={contest.id}>
              <H5>{contest.name}</H5>
              {contest.choices.map(choice => (
                <BlockLabel
                  key={choice.id}
                  htmlFor={`results[${contest.id}][${choice.id}]`}
                >
                  Votes for {choice.name}:
                  <Field
                    id={`results[${contest.id}][${choice.id}]`}
                    name={`results[${contest.id}][${choice.id}]`}
                    disabled={alreadySubmittedResults}
                    validate={testNumber()}
                    component={FormField}
                  />
                </BlockLabel>
              ))}
            </Card>
          ))}
          <BottomButton
            type="submit"
            intent="primary"
            disabled={alreadySubmittedResults}
            onClick={handleSubmit}
          >
            {alreadySubmittedResults
              ? `Already Submitted Data for Round ${round.roundNum}`
              : `Submit Data for Round ${round.roundNum}`}
          </BottomButton>
        </Form>
      )}
    </Formik>
  )
}

export default RoundDataEntry
