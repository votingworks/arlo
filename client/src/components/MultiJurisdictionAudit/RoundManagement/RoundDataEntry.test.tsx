import React from 'react'
import { render } from '@testing-library/react'
import RoundDataEntry from './RoundDataEntry'

describe('offline round data entry', () => {
  it('renders', () => {
    // TODO code offline data entry
    const { container } = render(<RoundDataEntry />)
    expect(container).toMatchSnapshot()
  })
})
