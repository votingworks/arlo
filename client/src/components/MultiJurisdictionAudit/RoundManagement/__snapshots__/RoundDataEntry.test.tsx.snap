// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offline round data entry renders 1`] = `
<div>
  <form>
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Round 
      1
       Data Entry
    </h2>
    <p>
      When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
    </p>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-1][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[contest-id-1][choice-id-1]"
              name="results[contest-id-1][choice-id-1]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-1][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[contest-id-1][choice-id-2]"
              name="results[contest-id-1][choice-id-2]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Contest 2
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-2][choice-id-3]"
      >
        Votes for 
        Choice Three
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[contest-id-2][choice-id-3]"
              name="results[contest-id-2][choice-id-3]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-2][choice-id-4]"
      >
        Votes for 
        Choice Four
        :
        <div
          class="sc-htpNat gdpVCF"
        >
          <div
            class="bp3-input-group sc-bxivhb eSUmfF"
          >
            <input
              class="bp3-input"
              id="results[contest-id-2][choice-id-4]"
              name="results[contest-id-2][choice-id-4]"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <button
      class="bp3-button bp3-intent-primary sc-dnqmqq bytmcb sc-htoDjs dESvso"
      type="submit"
    >
      <span
        class="bp3-button-text"
      >
        Submit Data for Round 1
      </span>
    </button>
  </form>
</div>
`;

exports[`offline round data entry submits 1`] = `
<div>
  <form>
    <h2
      class="bp3-heading sc-bdVaJa ibVpPq"
    >
      Round 
      1
       Data Entry
    </h2>
    <p>
      When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
    </p>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Contest 1
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-1][choice-id-1]"
      >
        Votes for 
        Choice One
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-1][choice-id-2]"
      >
        Votes for 
        Choice Two
        :
        2
      </label>
    </div>
    <div
      class="bp3-card bp3-elevation-2 sc-bwzfXH hrCotL"
    >
      <h5
        class="bp3-heading"
      >
        Contest 2
      </h5>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-2][choice-id-3]"
      >
        Votes for 
        Choice Three
        :
        1
      </label>
      <label
        class="sc-iwsKbI gesrUq"
        for="results[contest-id-2][choice-id-4]"
      >
        Votes for 
        Choice Four
        :
        2
      </label>
    </div>
    <button
      class="bp3-button bp3-disabled bp3-intent-primary sc-dnqmqq bytmcb sc-htoDjs dESvso"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="bp3-button-text"
      >
        Already Submitted Data for Round 1
      </span>
    </button>
  </form>
</div>
`;
